docs/index.bs

```diff
@@ -1986,78 +1983,64 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
       <dfn method for="Cache"><code>add(|request|)</code></dfn> method *must* run these steps:
 
         1. Let |requests| be an array containing only |request|.
-        1. Set |responseArrayPromise| to the result of running the algorithm specified in {{Cache/addAll(requests)}} passing |requests| as the argument.
-        1. Return the result of <a>transforming</a> |responseArrayPromise| with a fulfillment handler that returns undefined.
+        1. Let |responseArrayPromise| be the result of running the algorithm specified in {{Cache/addAll(requests)}} passing |requests| as the argument.
+        1. Return the result of [=transforming=] |responseArrayPromise| with a fulfillment handler that returns undefined.
     </section>
 
     <section algorithm="cache-addAll">
       <h4 id="cache-addAll">{{Cache/addAll(requests)}}</h4>
 
       <dfn method for="Cache"><code>addAll(|requests|)</code></dfn> method *must* run these steps:
 
-        1. Let |responsePromiseArray| be an empty array.
-        1. Let |requestArray| be an empty array.
+        1. Let |r| be null.
```

Comment:
I don't think we need this, see below.

Reply:
I intentionally factored the definition of |r| out of the nested blocks to reduce redundancy. It seems not a big deal as the local variable will likely to be stored in a register in machine level. But which is a better practice? I thought defining them within for-blocks is analogous to doing the same in implementations.

Reply:
Implementations may optimise by reusing variables, but I don't think it's something we should do in the spec as it makes it harder to read. The way it's currently scoped, I expected the second loop to use the current value of |r|. or |r| to be used after both loops.

I think variables should be as shallowly-scoped as possible, so if they're only used in a loop, then they can be scoped to that loop.

In this case, it means |r| will only ever be "let", never "set", so readers can tell it's a constant.

Reply:
Convinced. I'll change them back. Thanks!

Reply:
Addressed.